app/src/main/java/example/jackliu/nonewfriends/HomeActivity.java

```diff
@@ -87,6 +103,7 @@ public void onAuthStateChanged(@NonNull FirebaseAuth firebaseAuth) {
                     // user auth state is changed - user is null
                     // launch login activity
                     startActivity(new Intent(HomeActivity.this, LoginActivity.class));
+                    finish();
```

Comment:
@jackjunliu does this `finish()` belong here?

Reply:
It does because you want to close out all pages if user is not logged in.